### PR TITLE
[CPROD-925] IS20 refactor with the new cansiter-sdk api

### PR DESCRIPTION
**Pull request by itsyaasir**
The file, functions and variable names are subject to changes.

### CODE REVIEW

File: src/token/src/canister.rs [hunk 1]
```diff
@@ -185,228 +80,20 @@ impl TokenCanister {
...
-    /// Sets the minimum time between two consecutive auctions, in seconds.
-    ///
-    /// Only the owner is allowed to call this method.
-    #[update]
-    pub fn setAuctionPeriod(&self, period_sec: u64) -> Result<(), TxError> {
-        let caller = CheckedPrincipal::owner(&self.state.borrow_mut().stats)?;
-        // IC timestamp is in nanoseconds, thus multiplying
-        self.update_stats(caller, CanisterUpdate::AuctionPeriod(period_sec));
-        Ok(())
+    fn canister(&self) -> &TokenCanister {
```
Comment (awkure) [inline src/token/src/canister.rs:95]: Why is this needed?

File: src/token/src/canister/erc20_transactions.rs [hunk 2]
```diff
@@ -952,7 +958,8 @@ mod proptests {
     }
 
     prop_compose! {
-        fn make_tokens128() (num in "[0-9]{1,4}") -> Tokens128 {
+        fn make_tokens128() (num in "[0-9]{1,10}") -> Tokens128 {
+            eprintln!("{}", num);
```
Comment (awkure) [inline src/token/src/canister/erc20_transactions.rs:962]: remove

File: src/token/src/core.rs [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+use crate::canister::is20_notify::{approve_and_notify, consume_notification, notify};
+use crate::canister::is20_transactions::{batch_transfer, transfer_include_fee};
+use crate::canister::TokenCanister;
+use crate::principal::{CheckedPrincipal, Owner};
+use crate::state::CanisterState;
+use crate::types::{
+    AuctionInfo, Metadata, StatsData, TokenInfo, TxError, TxId, TxReceipt, TxRecord,
+};
+
+pub type AsyncReturn<'a, T> = Pin<Box<dyn Future<Output = T> + 'a>>;
```
Comment (awkure) [inline src/token/src/core.rs:26]: Why not importing it from canister-sdk directly?

Comment (itsyaasir) [reply src/token/src/core.rs:26]: Yeah, I haven't thought about it. On it

File: src/token/src/canister/erc20_transactions.rs (hunk 2 above)
Comment (itsyaasir) [reply src/token/src/canister/erc20_transactions.rs:962]: Ok

File: src/token/src/canister.rs (hunk 1 above)
Comment (itsyaasir) [reply src/token/src/canister.rs:95]: I was looking for a method on how to use the canister for the transfer methods.

File: src/token/src/core.rs [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    fn setAuctionPeriod(&self, period_sec: u64) -> Result<(), TxError> {
+        let caller = CheckedPrincipal::owner(&self.state().borrow_mut().stats)?;
+        // IC timestamp is in nanoseconds, thus multiplying
+        self.update_stats(caller, CanisterUpdate::AuctionPeriod(period_sec));
+        Ok(())
+    }
+
+    #[update(trait = true)]
+    fn consume_notification<'a>(&'a self, transaction_id: TxId) -> AsyncReturn<TxReceipt> {
+        // consume_notification(self, transaction_id)
```
[the remark this answers is not in the document]
Comment (itsyaasir) [reply src/token/src/core.rs:327]: 👍🏾

File: src/token/src/canister.rs (hunk 1 above)
Comment (awkure) [reply src/token/src/canister.rs:95]: Well, this is an `id` function, you can use `self` as is 🤔

Comment (itsyaasir) [reply src/token/src/canister.rs:95]: I don't understand this, what do you mean exactly ?

**Comment by awkure** [comment]
Also please remove `.DS_STORE` file

**Comment by Maximkaaa** [comment]
Oh, I see that my comments are addresed by #88 . It means that this PR can be closed.
